### PR TITLE
Tab titles change according to sound

### LINKED ISSUE #269: bug: tab titles do not update when leaving sound pages
Tab titles do not change when searching, or visiting the homepage through this application's logo, after leaving a sound's page.

To see this in action, visit a sound page, e.g., [Music Note 5's sound page](https://tide-app.vercel.app/sound/462807), or, [Shepard Note A's sound page](https://tide-app.vercel.app/sound/376626), and either search something, or visit the homepage by clicking on this application's logo. In both cases, the tab's title retains the title it had on the sound's page.

Instead, the tab title should say we are on a search results page, or on the home page of the application, respectively.

The fix should be straightforward. We can `import { Helmet } from "react-helmet";` to `App.js`, and add
```
                <Helmet>
                  <title>Tide - Explore a world of sounds</title>
                  <meta name="description" content="Homepage of Tide" />
                </Helmet>
```
to the `render` of the `<Route>` with `path={"/"}`, similar to what we've already done in `Sound.js`.

Similarly, we can `import { Helmet } from "react-helmet";` to `SearchResults.js`, and add
```
      {showHeader && (
        <Helmet>
          <title>{`Search results for "${searchQuery || searchValue}"`}</title>
          <meta name="description" content="A sound from Tide" />
        </Helmet>
      )}
```
to it; we only update the tab title if we are showing search results while not on the homepage, which happens when `showHeader` is truthy.

We can probably remove `<title>Tide - Explore a world of sounds</title>` from line 180 of `index.html` since `react-helmet` should cover all of our tab's titles after making the changes just mentioned.

To make an implicit change explicit, if we do implement the above changes, `SearchResults.js` would need to return a React Fragment since we would be returning more than one element.

image

image

### PR DESCRIPTION
**Pull request by akshara-sun**
fixes #269 
------
Tab titles now change according to whichever sound you navigate to!

Thanks again @hamirmahal for outlining the bug and solution!

### CODE REVIEW

File: src/SearchResults.js [hunk 1]
```diff
@@ -46,9 +47,14 @@ export default function Search({
   return (
     <SoundList
       header={
-        showHeader
-          ? `Search results for "${searchQuery || searchValue}"`
-          : undefined
+        showHeader && (
```
Comment (amilajack) [inline src/SearchResults.js:50]: the `header` prop expects a string. The `Helmet` component needs to be adjacent to `SoundList` like in this example:

```
<>
  <SoundList ... />
  <Helmet ... />
</>
```

**Review by amilajack — CHANGES_REQUESTED** [review]
Nice, almost there! Just a few fixes

File: src/SearchResults.js (hunk 1 above)
Comment (akshara-sun) [reply src/SearchResults.js:50]: Hold on, how can the` <Helmet>` prob be adjacent to `<Soundlist>` if it's inside of `<Soundlist>`? @amilajack

**Review by hamirmahal — CHANGES_REQUESTED** [review]
I added a couple of clarifying comments.

File: src/SearchResults.js [hunk 2]
```diff
@@ -46,9 +47,14 @@ export default function Search({
...
+          <Helmet>
+            <title>{`Search results for "${
+              searchQuery || searchValue
+            }"`}</title>
+            <meta name="description" content="A sound from Tide" />
```
[the remark this answers is not in the document]
Comment (akshara-sun) [reply src/SearchResults.js:55]: Thank you so much for the explanation @hamirmahal! Really helped me!

[the remark this answers is not in the document]
Comment (hamirmahal) [reply src/SearchResults.js]: You're welcome! Happy to help.

File: src/SearchResults.js [hunk 3]
```diff
@@ -45,19 +45,21 @@ export default function Search({
...
-      }
-      tracks={searchResults}
-      className="py-16"
-    />
+    <>
+      header=
+      {showHeader && (
+        <Helmet>
+          <title>{`Search results for "${searchQuery || searchValue}"`}</title>
+          <meta name="description" content="A sound from Tide" />
```
Comment (hamirmahal) [inline src/SearchResults.js:53]: I think we just need to change `content="A sound from Tide"` to `content="Search results from Tide"`. @amilajack might have more suggestions, but that's all I can think of currently.

**Review by hamirmahal — CHANGES_REQUESTED** [review]
I think we just need to change the `content` field of the `<meta />` tag in `src/SearchResults.js` from `content="A sound from Tide"` to `content="Search results from Tide"`.

File: src/SearchResults.js (hunk 3 above)
Comment (akshara-sun) [reply src/SearchResults.js:53]: Oops sorry! I missed that! Will push that change now

File: src/SearchResults.js [hunk 4]
```diff
@@ -44,14 +45,21 @@ export default function Search({
...
-      header={
-        showHeader
-          ? `Search results for "${searchQuery || searchValue}"`
-          : undefined
-      }
-      tracks={searchResults}
-      className="py-16"
-    />
+    <>
+      header=
```
Comment (amilajack) [inline src/SearchResults.js:49]: This part doesn't make sense

**Review by amilajack — CHANGES_REQUESTED** [review]

File: src/SearchResults.js (hunk 4 above)
Comment (akshara-sun) [reply src/SearchResults.js:49]: Could you please elaborate? I don't think I understand

Comment (amilajack) [reply src/SearchResults.js:49]: what do you expect this line to do?

**Review by hamirmahal — APPROVED** [review]
I think we're ready to merge `tab-title-fixes` into `master`.

One thing I'll note is that I think we should stay in the habit of keeping descriptive commit messages. Even if we squash commit messages and we don't see most of the individual commit messages in a pull request, it's an important habit to practice, in my opinion. The first three commits in this pull request at the time of this writing are fine, but we might be able to say something like "make header appear when returning search results" for the fourth commit.

**Review by amilajack — APPROVED** [review]

**Comment by amilajack** [comment]
Tide just keeps getting better day by day! 💪

**Pull request merged**